### PR TITLE
[Audit Logs] Don't log events before Strapi is loaded or if there is no user

### PR DESCRIPTION
**Pull request by Feranchz**
### What does it do?

In the `processEvent` function on Audit Logs service now we check if Strapi is loaded to log an event and also we check if there is a user executing the action, if not we don't log the event

### Why is it needed?

We are logging the events at a low level, so when Strapi starts it is logging too many permissions events that are not necessary to log, also there are some events without user logged in the audit logs (example: the content events from the API content)

### How to test it?

- Run a Strapi application with a valid EE license
- Create a new content type or edit a existing one and wait for Strapi to restart 
- Go to the audit logs page, you should see only the create/update event and not the delete permissions events

### CODE REVIEW

**Review by Convly — CHANGES_REQUESTED** [review]
Looks good to me, could you just add a new unit test that tries to call the audit log service with strapi.isLoaded being false?

**Review by remidej — COMMENTED** [review]
I think this is nice for the events at launch, but will not fix all the events created when making changes in the CTB. Did you look into the solution of not saving events that have no user?

**Review by Convly — COMMENTED** [review]
LGTM regarding the goal of this PR. I'll let you discuss with the rest of your squad if you want to modify its scope or not.

**Review by markkaylor — APPROVED** [review]
lgtm, I updated https://github.com/strapi/strapi/pull/15504 to remove the filters for null users

File: packages/core/admin/ee/server/services/audit-logs.js [hunk 1]
```diff
@@ -71,6 +75,11 @@ const createAuditLogsService = (strapi) => {
       return null;
     }
 
+    // Ignore events without a user
+    if (!strapi.requestContext.get()?.state?.user) {
```
Comment (Convly) [inline packages/core/admin/ee/server/services/audit-logs.js:79]: `ctx.state.user` can also represent an [authenticated U&P user](https://github.com/strapi/strapi/blob/main/packages/plugins/users-permissions/server/strategies/users-permissions.js#L51)
Do we want to also log events from those users?

**Review by remidej — COMMENTED** [review]
- As @Convly said, we need a way to exclude Content API users logged in via the users & permissions plugin. We're discussing it [on Slack](https://strapihq.slack.com/archives/C01S5EBKGAW/p1675251489301539)
- Since we now filter out events without a user, I don't know if it's necessary to wait for Strapi to be loaded, since the events happening before don't have a user

File: packages/core/admin/ee/server/services/audit-logs.js (hunk 1 above)
Comment (Convly) [reply packages/core/admin/ee/server/services/audit-logs.js:79]:
```suggestion
    const { isNil, propEq } = require('lodash/fp');

    const state = strapi.requestContext.get()?.state ?? {};

    const isUsingAdminAuthStrategy = propEq('strategy.name', 'admin', state);
    const hasAuthenticatedUser = !isNil(state.user);

    // Ignore events without an authenticated user using the admin auth strategy
    if (!hasAuthenticatedUser || !isUsingAdminAuthStrategy) {
      return;
    }
```

**Pull request closed without merging**